server/src/modules/users/application/use-cases/create-user/create-user-use-case.ts

```diff
@@ -52,8 +62,21 @@ export class CreateUserUseCase implements UseCaseWithDTO<CreateUserDTO, CreateUs
 
       const user = userResult.value
       await this.userRepo.save(user)
+      const updatedUser = await this.userRepo.getUserByUserEmail(email);
+      if(updatedUser.isErr())
+        return Result.err(new AppError.UnexpectedError(updatedUser.error.message))
+      
+        const tokenResponse = await new UserAuthHandler().create(updatedUser.value.id.toString())
+
+      if(tokenResponse.isErr())
+        return tokenResponse
+
+      const createUserSuccessResponse: CreateUserSuccess = {
+        user: UserMap.toDTO(user),
+        token: tokenResponse.value
+      }
```

Comment:
We need to update the use case unit test to test this new code, and all of its possible flows of control.